Update doc of listen-client-urls and listen-client-http-urls

Sadamingh opened the pull request:
PR for issue: #777 to - doc on listen-client-http-urls for gRPC and HTTP seperation.
Original discussions: : https://github.com/etcd-io/etcd/issues/15446

Sadamingh commented:
/cc @jmhbnz Thanks

jmhbnz commented on content/en/docs/v3.5/op-guide/configuration.md:
```diff
@@ -53,7 +53,9 @@ The list of flags provided below may not be up-to-date due to ongoing developmen
   List of URLs to listen on for peer traffic.
 --listen-client-urls 'http://localhost:2379'
-  List of URLs to listen on for client traffic.
+  List of URLs to listen on for client grpc traffic and http as long as --listen-client-http-urls is not specified.
```
We also need to include this flag in the `v3.4` and `v3.6` configuration pages.

Sadamingh replied:
Sure, will do

jmhbnz reviewed (commented):
Thanks for raising this @Sadamingh, looks good, just need to update the other docs versions mentioned below as well.

Sadamingh commented:
Update: added --listen-client-http-urls flag directions to doc v3.4 and v3.6
Modify: Changed default port of http urls to 2383 according to [etcd-io/etcd/tests/framework/e2e/cluster.go#L510](https://github.com/etcd-io/etcd/blob/74ca64116f5f73e8fd8e5ec68456ba74dbc87776/tests/framework/e2e/cluster.go#L510)
/cc @jmhbnz

jmhbnz commented on content/en/docs/v3.4/op-guide/configuration.md:
```diff
@@ -65,12 +65,20 @@ The list of flags provided below may not be up-to-date due to ongoing developmen
 
+### --listen-client-http-urls
++ List of URLs to listen on for HTTP-only client traffic. This flag tells the etcd to accept incoming requests from clients on the specified http://IP:port combinations. Enabling this flag removes HTTP services from --listen-client-urls. Use this flag when you want to segregate HTTP traffic from other protocols. If 0.0.0.0 is specified as the IP, etcd listens to the given port on all interfaces.  ...
++ default: "http://localhost:2383"
```
The default for this flag is `""`, refer:

```bash
 ➜ ./etcd --help    
  --listen-client-http-urls ''                                                                                                                               
    List of URLs to listen on for http only client traffic. Enabling this flag removes http services from --listen-client-urls. 
```

jmhbnz commented on content/en/docs/v3.5/op-guide/configuration.md:
```diff
@@ -53,7 +53,9 @@ The list of flags provided below may not be up-to-date due to ongoing developmen
 --listen-client-urls 'http://localhost:2379'
-  List of URLs to listen on for client traffic.
+  List of URLs to listen on for client grpc traffic and http as long as --listen-client-http-urls is not specified.
+--listen-client-http-urls 'http://localhost:2383'
```
As above, default is empty value.

jmhbnz commented on content/en/docs/v3.6/op-guide/configuration.md:
```diff
@@ -53,7 +53,9 @@ The list of flags provided below may not be up-to-date due to ongoing developmen
 --listen-client-urls 'http://localhost:2379'
-  List of URLs to listen on for client traffic.
+  List of URLs to listen on for client grpc traffic and http as long as --listen-client-http-urls is not specified.
+--listen-client-http-urls 'http://localhost:2383'
```
As above, please correct default.

jmhbnz reviewed (commented):
Thanks @Sadamingh - One last issue to fix for default value then this will be good to merge.

Sadamingh commented:
Thanks James, cleaned up default port for this flag.

jmhbnz reviewed (approved):
LGTM - Thanks @Sadamingh

ivanvc commented on content/en/docs/v3.4/op-guide/configuration.md:
```diff
@@ -65,12 +65,20 @@ The list of flags provided below may not be up-to-date due to ongoing developmen
 
 ### --listen-client-urls
-+ List of URLs to listen on for client traffic. This flag tells the etcd to accept incoming requests from the clients on the specified scheme://IP:port combinations. Scheme can be either http or https. Alternatively, use `unix://<file-path>` or `unixs://<file-path>` for unix sockets. If 0.0.0.0 is specified as the IP, etcd listens to the given port on all interfaces. If an IP address is given as  ...
++ List of URLs to listen on for client traffic. This flag tells the etcd to accept incoming requests from the clients on the specified scheme://IP:port combinations as long as --listen-client-http-urls is not specified. Scheme can be either http or https. Alternatively, use `unix://<file-path>` or `unixs://<file-path>` for unix sockets. If 0.0.0.0 is specified as the IP, etcd listens to the given ...
```
Suggestion to add code formatting to some elements, like:

* `scheme://IP:port`
* `--listen-client-http-urls`
* `0.0.0.0`
* `http`/`https`

```suggestion
+ List of URLs to listen on for client traffic. This flag tells the etcd to accept incoming requests from the clients on the specified `scheme://IP:port` combinations as long as `--listen-client-http-urls` is not specified. Scheme can be either `http` or `https`. Alternatively, use `unix://<file-path>` or `unixs://<file-path>` for unix sockets. If `0.0.0.0` is specified as the IP, etcd listens to the given port on all interfaces. If an IP address is given as well as a port, etcd will listen on the given port and interface. Multiple URLs may be used to specify a number of addresses and ports to listen on. The etcd will respond to requests from any of the listed addresses and ports.
```

ivanvc commented on content/en/docs/v3.4/op-guide/configuration.md:
```diff
@@ -65,12 +65,20 @@ The list of flags provided below may not be up-to-date due to ongoing developmen
 + invalid example: "http://example.com:2379" (domain name is invalid for binding)
 
+### --listen-client-http-urls
++ List of URLs to listen on for HTTP-only client traffic. This flag tells the etcd to accept incoming requests from clients on the specified http://IP:port combinations. Enabling this flag removes HTTP services from --listen-client-urls. Use this flag when you want to segregate HTTP traffic from other protocols. If 0.0.0.0 is specified as the IP, etcd listens to the given port on all interfaces.  ...
```
Same comment here.

```suggestion
+ List of URLs to listen on for HTTP-only client traffic. This flag tells the etcd to accept incoming requests from clients on the specified `http://IP:port` combinations. Enabling this flag removes HTTP services from `--listen-client-urls`. Use this flag when you want to segregate HTTP traffic from other protocols. If `0.0.0.0` is specified as the IP, etcd listens to the given port on all interfaces. If an IP address is given as well as a port, etcd will listen on the given port and interface. Multiple URLs may be used to specify a number of addresses and ports to listen on. The etcd will respond to requests from any of the listed addresses and ports.
```

ivanvc reviewed (commented):
Thanks, @Sadamingh. This looks great; I left two minor possible improvements.

Sadamingh commented:
Thanks @ivanvc, these code formattings are added.
/cc @spzala

spzala reviewed (approved):
Thanks @Sadamingh lgtm!

ivanvc reviewed (approved):
LGTM. Thanks, @Sadamingh
